### ISSUE TITLE
OrderedMethods fails to scan classes that have siblings

### ISSUE THREAD

**Issue by Darhazer** [shanecav84/rubocop-ordered_methods#8]
Given the code
```
class A
  class B
    def foo
    end
    def baz
    end
  end

  def b
  end

  def a
  end
end
```

OrderedMethods ignores the offences in class B

Given the code
```
require 'some_lib'

class A
  def b
  end

  def a
  end
end
```
OrderedMethods doesn't find the offences in class A. (note that removing the require fixes the issue - the cop detects that `a` has to be before `b`.

I think this is caused by `cache` method, which incorrectly caches siblings from previous `begin` node. It seems that the `on_begin` is being called twice, with the two `begin` nodes in the examples above, but on the same cop instance.

**Comment by shanecav84** [comment]
Thanks for the issue. I'll take a look this week.

**Comment by shanecav84** [comment]
Should be fixed by #15

**Comment by Darhazer** [comment]
It seems that now it detects the offence only in the inner class, not the outer:
Before (version 0.10):
```
test.rb:12:3: C: [Correctable] Layout/OrderedMethods: Methods should be sorted in alphabetical order.
  def a ...
  ^^^^^
```

After (version 0.11)
```
test.rb:5:5: C: [Correctable] Layout/OrderedMethods: Methods should be sorted in alphabetical order.
    def baz ...
    ^^^^^^^
```
